monai/visualize/occlusion_sensitivity.py

```diff
@@ -130,168 +145,170 @@ class OcclusionSensitivity:
         num_required_predictions = np.prod(downsampled_im_shape)
 
+        # Get bottom left and top right corners of occluded region
+        lower_corner = (self.stride - self.mask_size) // 2
+        upper_corner = (self.stride + self.mask_size) // 2
+
         # Loop 1D over image
         verbose_range = trange if self.verbose else range
         for i in verbose_range(num_required_predictions):
             # Get corresponding ND index
             idx = np.unravel_index(i, downsampled_im_shape)
             # Multiply by stride
-            idx *= stride_np
+            idx *= self.stride
             # If a bounding box is being used, we need to add on
             # the min to shift to start of region of interest
             if b_box_min is not None:
                 idx += b_box_min
 
-            # Get min and max index of box to occlude
-            min_idx = [max(0, i - self.margin) for i in idx]
-            max_idx = [min(j, i + self.margin) for i, j in zip(idx, im_shape)]
+            # Get min and max index of box to occlude (and make sure it's in bounds)
+            min_idx = np.maximum(idx + lower_corner, 0)
+            max_idx = np.minimum(idx + upper_corner, im_shape)
 
             # Clone and replace target area with `pad_val`
             occlu_im = x.detach().clone()
-            occlu_im[(...,) + tuple(slice(i, j) for i, j in zip(min_idx, max_idx))] = self.pad_val
+            occlu_im[(...,) + tuple(slice(i, j) for i, j in zip(min_idx, max_idx))] = pad_val
 
             # Add to list
             batch_images.append(occlu_im)
-            batch_ids.append(class_idx)
 
             # Once the batch is complete (or on last iteration)
             if len(batch_images) == self.n_batch or i == num_required_predictions - 1:
-                # Do the predictions and append to sensitivity map
-                sensitivity_im = _append_to_sensitivity_im(self.nn_module, batch_images, batch_ids, sensitivity_im)
+                # Do the predictions and append to sensitivity maps
+                sensitivity_ims = _append_to_sensitivity_ims(self.nn_module, batch_images, sensitivity_ims)
                 # Clear lists
                 batch_images = []
-                batch_ids = []
-
-        # Subtract baseline from sensitivity so that +ve values mean more important in decision process
-        sensitivity_im = baseline - sensitivity_im
 
         # Reshape to match downsampled image, and unsqueeze to add batch dimension back in
-        sensitivity_im = sensitivity_im.reshape(tuple(downsampled_im_shape)).unsqueeze(0)
+        for i in range(num_classes):
+            sensitivity_ims[i] = sensitivity_ims[i].reshape(tuple(downsampled_im_shape)).unsqueeze(0)
 
-        return sensitivity_im, output_im_shape
+        return sensitivity_ims, output_im_shape
 
     def __call__(  # type: ignore
-        self, x: torch.Tensor, class_idx: Optional[Union[int, torch.Tensor]] = None, b_box: Optional[Sequence] = None
+        self,
```

Comment:
I thought with non-empty `class_idx` option it'll save some memory here?

Reply:
There's a marginal increase in memory, the output image might be e.g., `64,64,64,3` instead of `64,64,64` for 3D and even smaller for 2D. Since batch size is always 1, the images will always be small, and it seems a shame to discard the information when it's generated anyway during the inference step at no extra cost.